### ISSUE TITLE
docker-compose*.yml : bump versions of postgres and elasticsearch

### ISSUE THREAD

**Issue by benosteen** [archesproject/arches#4753]
**Describe the bug**
`docker-compose*.yml` specifies old versions for postgres (9.6) and elasticsearch(5.2), which are out of sync with the master branch dependencies.

**Expected behavior**
`docker-compose*.yml` should reference:

* `elasticsearch:6.6.2` (rather than 5.2)
* (Optionally) `mdillon:postgis:11-alpine`

**Additional context**
The `docker-compose-test.yml` stack is run when using the automated build system for Docker Hub (and potentially other build systems), so would be useful for this to be in line with `master`

**Comment by benosteen** [comment]
It may also require the Dockerfile to include updated client libraries too.

**Comment by benosteen** [comment]
Latest master has shifted to ES 6.7.1 as per #4757

**Comment by bptarpley** [comment]
i'm trying to deploy a new instance of arches (this is my first time) via docker. using the docker-compose.yml file in this repo, there's a problem when first setting up the database, ie running

`python manage.py packages -o setup_db`

specifically, the elasticsearch container throws this error:

`elasticsearch.exceptions.RequestError: TransportError(400, u'illegal_argument_exception', u'Rejecting mapping update to [strings] as the final mapping would have more than 1 type: [concept, term]')`

this seems like an incompatability with newer (> 6) version of elasticsearch. please advise.

**Comment by benosteen** [comment]
What version of Arches are you using or are you using a particular branch
on the github repo?

**Comment by bptarpley** [comment]
i'm using the version specified in the docker-compose file (archesproject/arches:latest).

**Comment by benosteen** [comment]
Looking at their dockerhub, the archesproject images haven't been updated
for a number of months. The 'latest' tagged image was uploaded 8 months
ago, and likely corresponds to 4.3.3. If you are only using the
docker-compose.yml file and nothing else, you could have some issues.

If you are instead downloading the repository and attempting to use the
docker-compose.yml file, you may have noticed that it will go through a
build step where it attempts to build am 'archesproject/arches:latest'
tagged container from the source code. Are you using a tagged release, or
did you download the latest code from the 'master' branch of Arches?

Ben

**Comment by bptarpley** [comment]
ok. have now cloned the master branch of repo and am attempting to build the container. when it gets to line 133, it fails to find package.json in the root of the repo.

**Comment by benosteen** [comment]
I'm sorry to say that while you are now up to date, the master branch is
broken. I had filed a pull request to try fix this, but it is still waiting
for approval to be pushed to master:

https://github.com/archesproject/arches/pull/4882

Ben

**Comment by benosteen** [comment]
I have also filed a new high priority issue as this error is also present
in the new 4.4.2 release:
https://github.com/archesproject/arches/issues/4971

**Comment by bptarpley** [comment]
thanks, ben! will check back next week to see if these issues are resolved :)

**Comment by bptarpley** [comment]
one last thing--it also seems that certain default databases need to be created for the CouchDB instance. when firing up the CouchDB container, it constantly complains, for instance, that there is no _users database.

**Comment by benosteen** [comment]
The setup should create those in the CouchDB container, but I had a look at
the default docker-compose.yml and it is attempting to persist the couchdb
data volume with the incorrect folder path (IMO). I think the volume
configuration should be "couchdb-data:/opt/couchdb/data", rather than
"couchdb-data:/usr/local/var/lib/couchdb".
I don't use the CouchDB component, as it only relates to the mobile
surveying component of Arches and this service can be disabled without
affecting Arches (if you are not using that mobile survey upload feature
yourself).

In short, my assumption is that the setupdb command would create the
CouchDB _users, etc points, but as soon as the container is spun down or
restarted, that information has gone (which will create the errors). It may
also be broken in some other way instead but this is my hunch!

Ben
